var/spack/repos/builtin/packages/fenics-dolfinx/package.py

```diff
@@ -42,12 +44,16 @@ class FenicsDolfinx(CMakePackage):
 
     depends_on("py-fenics-ffcx", type=("build", "run"))
     depends_on("py-fenics-ffcx@main", type=("build", "run"), when="@main")
+    depends_on("py-fenics-ffcx@0.4.1", type=("build", "run"), when="@0.4.1")
+    depends_on("py-fenics-ffcx@0.4.0", type=("build", "run"), when="@0.4.0")
     depends_on("py-fenics-ffcx@0.3.0", type=("build", "run"), when="@0.3.0")
     depends_on("py-fenics-ffcx@0.2.0", type=("build", "run"), when="@0.2.0")
     depends_on("py-fenics-ffcx@0.1.0", type=("build", "run"), when="@0.1.0")
```

Comment:
FYI. You could actually use a for loop to set up these version-specific dependencies here and below, e.g.:

```suggestion
    for ver in ['0.1.0', '0.2.0', '0.3.0', '0.4.0', '0.4.1']:
        depends_on("py-fenics-ffcx@{0}".format(ver), type=("build", "run"), when="@{0}".format(ver))
```

Reply:
You could actually include `main` as a version for that matter.

Reply:
This is a good suggestion. If other revisions are necessary I will include this as well.